modules/programs/readline.nix

```diff
@@ -0,0 +1,44 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let
+  cfg = config.programs.readline;
+in
+{
+  options.programs.readline = {
+    enable = mkEnableOption "readline";
+
+    bindings = mkOption {
+      default = {};
+      type = types.attrsOf types.str;
+      example = { "\C-h" = "backward-kill-word"; };
+      description = "Readline bindings";
+    };
+
+    includeSystem = mkOption {
+      type = types.bool;
+      default = true;
+      description = "Whether to include the system-wide config";
+    };
+
+    extraConfig = mkOption {
+      type = types.lines;
+      default = "";
+      description = "Config lines appended to the end unchanged. Useful for configuration not yet supported by home-manager.";
+    };
+  };
+  config = mkIf cfg.enable {
+    home.packages = [ pkgs.readline ];
+    home.file.".inputrc".text =
+    let
+      includeSystemStr = if cfg.includeSystem then "$include /etc/inputrc" else "";
+      bindingsStr = concatStringsSep "\n" (
+        mapAttrsToList (k: v: "\"${k}\": ${v}") cfg.bindings
+      );
+    in
+      ''
+        ${includeSystemStr}
+        ${bindingsStr}
+        ${cfg.extraConfig}
+      '';
```

Comment:
Need to be indented two spaces. Also could use `optionalString`:
```nix
    let
      includeSystemStr =
        optionalString cfg.includeSystemConfig "$include /etc/inputrc";

      bindingsStr = concatStringsSep "\n" (
        mapAttrsToList (k: v: "\"${k}\": ${v}") cfg.bindings
      );
    in
      ''
        ${includeSystemStr}
        ${bindingsStr}
        ${cfg.extraConfig}
      '';
```

Reply:
I am sorry, I am not sure what you mean here

Reply:
About the indentation you now have
```
    home.file.".inputrc".text =
    let
```
and I'd like to have
```
    home.file.".inputrc".text =
      let
```
and about `optionalString` instead of
```
if cfg.includeSystem then "$include /etc/inputrc" else ""
```
you can have
```
optionalString cfg.includeSystem "$include /etc/inputrc"
```
```